File: BaseLibrary/communication/command/common/DriveAnswerCommand.cs
using System;
using System.Collections.Generic;

namespace BaseLibrary.communication.command.common {
    /// <summary>
    /// Answer for drive command.
    /// </summary>
    public class DriveAnswerCommand : ACommonCommand {
        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        private bool _success;

        /// <summary>
        /// True if robot change direction.
        /// </summary>
        public bool SUCCESS {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _success;
            }
            private set => _success = value;
        }

        public DriveAnswerCommand() {}

        public DriveAnswerCommand(bool success) {
            SUCCESS = success;
            pending = false;
        }

        public void FillData(DriveAnswerCommand source) {
            SUCCESS = source.SUCCESS;
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/common/DriveAnswerCommand.java

package BaseLibrary.communication.command.common;

import BaseLibrary.communication.protocol.AProtocol;

/**
 * Answer for drive command.
 */
public class DriveAnswerCommand extends ACommonCommand {

	private static final String NAME = "DRIVE_ANSWER";
	
	static{
		AProtocol.registerForDeserialize(NAME, DriveAnswerCommand.class);
	}
	
	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private boolean success;

		public DriveAnswerCommand() {
	}

	public DriveAnswerCommand(boolean success) {
		setSuccess(success);
	}

	/**
	 * Fill command by data from another command (use-full for filling pending
	 * command by command with data).
	 */
	public void FillData(DriveAnswerCommand source) {
		setSuccess(source.success);
	}

	/**
     * True if robot change direction.
     */
	public boolean isSuccess() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return success;
	}

	private void setSuccess(boolean success) {
		pending = false;
		this.success = success;
	}
}
